File: 162-find-peak-element/find-peak-element.cpp
class Solution {
public:
    int findPeakElement(vector<int>& nums) {
        int l =0, r =nums.size()-1;
        int ret = -1;
        while (l<r){
            int m = (r+l)/2;
            if(nums[m]>nums[m+1]){
                r = m;
            }
            else {
                l = m+1;
            }
        }

        return l;
    }
};

File: 162-find-peak-element/find-peak-element.py
class Solution:
    def findPeakElement(self, nums: List[int]) -> int:
        l,r= 0,len(nums)
        while(l<=r):
            mid = (r+l)//2
            left = nums[mid-1] if mid-1>=0 else float("-inf")
            right = nums[mid+1] if mid+1<len(nums) else float("-inf")
            if left<=nums[mid] and right<=nums[mid]:
                return mid
            elif right>nums[mid]:
                l = mid+1
            else:
                r = mid-1
        return l